transcom/mymove
# [MB-16137] Integrate new service item card

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16137)

## Summary

This replaces the old ServiceItemCards and ReviewServiceItems components with their updated versions, integrates them with the PaymentRequestReview, and updates all related tests.

The end result is that, when rejecting a service item card, the `rejectionReason` field is required. The "Save" button that previously saved the reason is removed, and the "Next" button now takes care of validating, submitting, and navigating from the service item review card. When the user clicks "Next" with Reject selected and no reason, a validation appears and navigation is prevented. 

Note that the majority of the code changes here had been previously tested in the `NewServiceItemCard` and `NewReviewServiceItems` components; in general these have just been brought in to replace the existing versions. The relevant changes are all in the `PaymentRequestReview` and its tests, and in the `tioFlows` integration tests. Any Happo diffs should just be renames.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the office app
2. Login as a TIO
3. Select a move with "Payment Requested" status
4. Click "Review service items"
5. Reject an item without providing a reason, and click "Next"
6. Note the error state
7. Provide a reason and proceed
8. Ensure that the rest of the flow is as expected

### Frontend
